sklearn/neighbors/tests/test_kde.py

```diff
@@ -141,3 +141,44 @@ def test_kde_pipeline_gridsearch():
     search = GridSearchCV(pipe1, param_grid=params, cv=5)
     search.fit(X)
     assert_equal(search.best_params_['kerneldensity__bandwidth'], .1)
+
+
+def test_kde_sample_weights():
+    N = 10000
```

Comment:
Please make sure the test is not too long, reducing `N` and `T` if necessary.
You can do it with `pytest --duration 10 sklearn/neighbors/tests/test_kde.py::test_kde_sample_weights`.